tests/integration/test_action.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration test for actions."""
+
+import pytest
+from juju.application import Application
+
+from .conftest import TEST_EXTERNAL_HOSTNAME_CONFIG
+
+
+@pytest.mark.abort_on_fail
+async def test_get_certificate_action(
+    configured_application_with_tls: Application,
+):
+    """
+    arrange: Deploy the charm with valid config and tls integration.
+    act: Run the get-certificate action and run a sh command to check
+    the cert location on the unit.
+    assert: The output of both operations are valid.
+    """
+    action = await configured_application_with_tls.units[0].run_action(
+        "get-certificate", hostname=TEST_EXTERNAL_HOSTNAME_CONFIG
+    )
+    await action.wait()
+    assert action.results
```

Comment:
What about checking the results content and also adding failure scenarios for example if you run the action without the relation?

Reply:
Thx, I added a failure case and also update the test to test the correct output.